src/middlewared/middlewared/plugins/device_/device_info.py

```diff
@@ -226,14 +226,18 @@ def safe_retrieval(self, prop, key, default, asint=False):
         return default
 
     @private
-    def get_disk(self, name):
+    def get_disk(self, name, get_partitions=False, serial_only=False):
         context = pyudev.Context()
         try:
             block_device = pyudev.Devices.from_name(context, 'block', name)
+            if serial_only:
+                return {name: self.get_disk_serial(block_device)}
```

Comment:
Btw if we are only retrieving disk here, should we be adding `name` to the dict at all ?

Reply:
It kind of follows the returned object if serial_only=False

Reply:
Sure, my idea was that if it's only 1 disk which this endpoint is going to give data of - probably no need to use that name in the dict at all and just return the information directly or use `{"serial": value}` here

Reply:
That's a good idea. I changed it to return `{'serial': ''}`